ipaclient/install/client.py

```diff
@@ -1275,7 +1275,7 @@ def do_nsupdate(update_txt):
 
     update_fd = open(UPDATE_FILE, "w")
     update_fd.write(update_txt)
-    update_fd.flush()
+    ipautil.flush_sync(update_fd)
```

Comment:
While you are on it, please also refactor and fix simple file operations to use a context manager:

```
with open(UPDATE_FILE, "w") as f:
    f.write(update_txt)
    ipautil.flush_sync(f)
```

Be careful when it's a temporary file. close() also deleted the file.